src/sentry/rules/processing/processor.py

```diff
@@ -302,6 +334,39 @@ def apply_rule(self, rule: Rule, status: GroupRuleStatus) -> None:
                 )
                 return
 
+        predicate_func = get_match_function(condition_match)
+        if condition_list:
+            if not predicate_func:
```

Comment:
I think we should only error here if `not fast_conditions and not slow_conditions`

Reply:
isn't this basically `if not condition list`? It's ok if there are no conditions, it shouldn't error

Reply:
Right, but we don't want to error if there are no conditions. In some cases, we only have `slow_conditions` and no conditions in `condition_list`, so we'd want to error in that case I think?

Reply:
in that case we just want to go straight to enqueueing the rule, right?

Reply:
If the `predicate_func is None` we'd still want to error if we had `slow_conditions` wouldn't we?

Reply:
Yes, I can pull that out to be before the if statement